src/components/Medications.tsx

```diff
@@ -1,13 +1,17 @@
 'use client';
+
 import React, { useState, useEffect } from 'react';
 import { supabase } from '@/services/supabase';
+import { Loader } from '@/components/Loader';
 
 interface MedicationsProps {
   pet_id: string;
 }
 
 export const Medications: React.FC<MedicationsProps> = ({ pet_id }) => {
   const [medications, setMedications] = useState<any[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
   const fetchMedications = async () => {
     const { data, error } = await supabase
       .from('pet_medications')
```

Comment:
🔴 let's use throw error instead of only showing it in console :)  so that we have better error handling execution
` if (error) {
        throw error;
      }`